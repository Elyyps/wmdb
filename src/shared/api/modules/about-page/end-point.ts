import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
