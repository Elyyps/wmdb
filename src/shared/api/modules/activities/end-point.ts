import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
