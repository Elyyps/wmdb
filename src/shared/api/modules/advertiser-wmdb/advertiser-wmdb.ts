export interface IAdvertiserWMDB {
  content: string;
  formTitle: string;
  title: string;
}
