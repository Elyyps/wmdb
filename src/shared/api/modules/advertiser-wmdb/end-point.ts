import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
