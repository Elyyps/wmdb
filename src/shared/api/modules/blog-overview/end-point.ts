import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
