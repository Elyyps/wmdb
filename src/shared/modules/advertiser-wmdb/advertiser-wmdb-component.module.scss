.advertiser-wmdb {
  ul {
    font-size: 16px;
    color: #101010;
    @media (max-width: $breakpoint-medium) {
      font-size: 14px;
    }
  }
  li {
    font-weight: bold;
  }

  p {
    margin: 8px 26px 24px 26px;
  }
}
