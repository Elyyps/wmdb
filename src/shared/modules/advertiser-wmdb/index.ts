export * from './advertiser-wmdb.component'