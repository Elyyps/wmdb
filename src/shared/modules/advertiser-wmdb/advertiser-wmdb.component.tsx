import * as React from "react";
import styles from "./advertiser-wmdb-component.module.scss";
import { ModuleSectionComponent } from "@app/core/module-section";
import { FAQFormComponent } from "@app/core/faq-form";
import { IAdvertiserWMDB } from "@app/api/modules/advertiser-wmdb/advertiser-wmdb";
import { BodyTextComponent } from "@app/core/bodytext";

export interface IAdvertiserWMDBComponentProps {
  advertiserModule: IAdvertiserWMDB;
}

const AdvertiserWMDBComponent = ({ advertiserModule }: IAdvertiserWMDBComponentProps) => (
  <ModuleSectionComponent
    paddingTop={"72px"}
    paddingBottom={"72px"}
    paddingTopMobile={"32px"}
    paddingBottomMobile={"32px"}
    pattern
  >
    <div className={"uk-grid uk-child-width-1-2@s uk-grid-xlarge"}>
      <div className={styles["advertiser-wmdb"]}>
        <h2>{advertiserModule.title}</h2>
        <BodyTextComponent html={advertiserModule.content} style={{ color: "#101010" }} />
        <br />
      </div>

      <div className={styles["advertiser-wmdb-form"]}>
        <h2>{advertiserModule.formTitle}</h2>
        <FAQFormComponent onSubmit={() => alert("submitted ")} />
      </div>
    </div>
  </ModuleSectionComponent>
);

export { AdvertiserWMDBComponent };
