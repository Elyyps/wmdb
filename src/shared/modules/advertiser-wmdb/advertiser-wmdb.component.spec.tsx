import React from "react";
import { shallow } from "enzyme";
import { AdvertiserWMDBComponent } from "./advertiser-wmdb.component";
import { generateAdvertiserData } from "@app/api/modules/advertiser-wmdb/end-point";

describe("[AdvertiserWMDB]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<AdvertiserWMDBComponent advertiserModule={generateAdvertiserData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
