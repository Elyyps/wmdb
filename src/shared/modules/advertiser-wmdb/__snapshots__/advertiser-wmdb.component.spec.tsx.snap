// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AdvertiserWMDB] should render component without crashing 1`] = `
<ModuleSectionComponent
  paddingBottom="72px"
  paddingBottomMobile="32px"
  paddingTop="72px"
  paddingTopMobile="32px"
  pattern={true}
>
  <div
    className="uk-grid uk-child-width-1-2@s uk-grid-xlarge"
  >
    <div
      className="advertiser-wmdb"
    >
      <h2>
        Waarom bij ons adverteren
      </h2>
      <BodyTextComponent
        html="<span>Door Wegmetdebaas.nl als marketingkanaal aan te spreken vergroot u de regio waarin uw klanten zich bevinden. </span>
  <ul className=\\"uk-list-check\\">
<li>Groot bereik onder vele bedrijven</li>
<p>Via onze hoge posities bij Google en groot bereik via onze nieuwsbrief Gooves en via Facebook</p>
<li>Uitgebreide presentatie mogelijkheid</li>
<p>Uw teksten, foto’s en filmpjes perfect online gespresenteerd, ook op tablet en mobiel</p>
<li>Beheer zelf de inhoud van uw advertentie</li>
<p>Altijd mogelijk om zelf teksten en foto’s aan te passen</p>
<li>3 Maandelijkse rapportage</li>
<p>Uitgebreide rapportage van uw marketingactie(s)</p>
</ul> "
        style={
          Object {
            "color": "#101010",
          }
        }
      />
      <br />
    </div>
    <div
      className="advertiser-wmdb-form"
    >
      <h2>
        Graag ontvang ik meer informatie over adverteren op WegmetdeBaas.nl
      </h2>
      <WithFormik(InnerForm)
        onSubmit={[Function]}
      />
    </div>
  </div>
</ModuleSectionComponent>
`;
